gravitee-apim-console-webui/src/management/settings/api-portal-header/migrated/api-portal-header.component.ts

```diff
@@ -0,0 +1,255 @@
+    this.portalSettingsService.save(this.settings)
+      .pipe(
+        tap(() => {
+          this.snackBarService.success("Promoted API is now " + (this.settings.portal.apis.promotedApiMode.enabled ? "visible" : "hidden"));
+        }),
+        catchError(({ error }) => {
+          this.snackBarService.error(error.message);
+          return EMPTY;
+        }),
+        takeUntil(this.unsubscribe$)
+      ).subscribe((updatedSettings: PortalSettings) => {
+      this.settings = updatedSettings;
+    });
+  };
+
+  public isReadonlySetting(property: string): boolean {
+    return PortalSettingsService.isReadonly(this.settings, property);
+  };
+
+  private getList(): void {
+    this.environmentApiHeadersService
+      .getApiHeaders()
+      .pipe(takeUntil(this.unsubscribe$))
+      .subscribe();
+  }
+
+  public moveUp(apiPortalHeader: ApiPortalHeader): void {
+    const updatedHeader = { ...apiPortalHeader, order: apiPortalHeader.order - 1 };
+    this.environmentApiHeadersService
+      .updateApiHeader(updatedHeader)
+      .pipe(takeUntil(this.unsubscribe$))
+      .subscribe();
+  }
+
+  public moveDown(apiPortalHeader: ApiPortalHeader): void {
+    const updatedHeader = { ...apiPortalHeader, order: apiPortalHeader.order + 1 };
+    this.environmentApiHeadersService
+      .updateApiHeader(updatedHeader)
+      .pipe(takeUntil(this.unsubscribe$))
+      .subscribe();
+  }
+
+  public createHeader(): void {
+    this.matDialog.open(HeaderDialogComponent, {
+      width: GIO_DIALOG_WIDTH.SMALL,
+      data: { name: "", value: "", isUpdate: false }
+    })
+      .afterClosed()
+      .pipe(
+        filter((data): boolean => !!data),
+        switchMap((data: DialogData) => this.environmentApiHeadersService.createApiHeader(data)),
+        takeUntil(this.unsubscribe$)
+      )
+      .subscribe();
+  }
+
+  public updateHeader(header: ApiPortalHeader): void {
+    this.matDialog.open(HeaderDialogComponent, {
+      data: { name: header.name, value: header.value, isUpdate: true },
+      width: GIO_DIALOG_WIDTH.SMALL
+    })
+      .afterClosed()
+      .pipe(
+        filter((data): boolean => !!data),
+        switchMap((data: DialogData) => {
+            const payload: ApiPortalHeader = {
+              ...header,
+              name: data.name,
+              value: data.value
+            };
+            return this.environmentApiHeadersService.updateApiHeader(payload);
+          }
+        ),
+        tap(() => {
+          this.snackBarService.success("Headers list updated.");
+        }),
+        catchError(({ error }) => {
+          this.snackBarService.error(error.message);
+          return EMPTY;
+        }),
+        takeUntil(this.unsubscribe$)
+      )
+      .subscribe();
+  }
+
+  public deleteHeader(header: ApiPortalHeader): void {
+    this.matDialog
+      .open<GioConfirmDialogComponent, GioConfirmDialogData, boolean>(GioConfirmDialogComponent, {
+        data: {
+          title: "",
```

Comment:
WDYT to configure with this kid of UI : 
![image](https://github.com/gravitee-io/gravitee-api-management/assets/4974420/feabb9bc-0d61-49ec-8a29-65382f2b01f7)